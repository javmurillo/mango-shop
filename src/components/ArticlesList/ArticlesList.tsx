import styled from 'styled-components';
import * as ArticleTypes from '../../store/types/articles.types';
import { Article } from '../Article/Article';
import { ArticlesListProps } from './articles-list-props.interface';

const StyledDiv = styled.div`
  justify-content: center;
`;

export const ArticlesList = (props: ArticlesListProps) => {
  return (
    <StyledDiv className="row">
      {props.articlesList.map((article: ArticleTypes.Article) => {
        return <Article article={article} key={article.id} />;
      })}
    </StyledDiv>
  );
};
