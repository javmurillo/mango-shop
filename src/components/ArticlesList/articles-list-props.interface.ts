import { Article } from '../../store/types/articles.types';

export interface ArticlesListProps {
  articlesList: Article[];
}
