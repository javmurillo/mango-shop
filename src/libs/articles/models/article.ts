export interface Article {
  id: string;
  name: string;
  description: string;
  img: string;
  currentPrice: number;
  oldPrice?: number;
}
