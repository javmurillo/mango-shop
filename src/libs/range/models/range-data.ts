export interface RangeData {
  min: number;
  max: number;
  rangeValues: number[];
}
