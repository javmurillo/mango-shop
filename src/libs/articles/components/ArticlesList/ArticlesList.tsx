import { Alert } from 'react-bootstrap';
import styled from 'styled-components';
import { Article as ArticleDto } from '../../models/article';
import { Article } from '../Article/Article';
import { ArticlesListProps } from './articles-list-props.interface';

const StyledDiv = styled.div`
  justify-content: center;
`;

const StyledAlert = styled(Alert)`
  margin-top: 2rem;
`;

/**
 * ArticlesList component which iterates over the props to show one card per article.
 * @param props ArticlesListProps
 */
export const ArticlesList = (props: ArticlesListProps): JSX.Element => {
  if (props.articlesList.length === 0) {
    return (
      <StyledAlert variant="dark">
        There is no articles to show matching the given filters!
      </StyledAlert>
    );
  }
  return (
    <StyledDiv className="row">
      {props.articlesList.map((article: ArticleDto) => {
        return <Article article={article} key={article.id} />;
      })}
    </StyledDiv>
  );
};
