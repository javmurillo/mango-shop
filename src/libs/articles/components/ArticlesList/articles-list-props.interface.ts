import { Article } from '../../models/article';

export interface ArticlesListProps {
  articlesList: Article[];
  ariaLabel?: string;
}
