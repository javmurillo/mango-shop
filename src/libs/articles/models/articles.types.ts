import { Action } from 'redux';
import { Article } from './article';

export const SET_ARTICLES = 'SET_ARTICLES';
export const FETCH_ARTICLES_FAILED = 'FETCH_ARTICLES_FAILED';
export const FILTER_ARTICLES = 'FILTER_ARTICLES';

export interface SetArticlesAction extends Action {
  type: typeof SET_ARTICLES;
  payload: { articles: Article[] };
}

export interface FetchArticlesFailedAction extends Action {
  type: typeof FETCH_ARTICLES_FAILED;
}

export interface FilterArticlesAction extends Action {
  type: typeof FILTER_ARTICLES;
  payload: { min: number; max: number };
}

export type ArticleAction =
  | SetArticlesAction
  | FetchArticlesFailedAction
  | FilterArticlesAction;

export type ArticleDispatchType = (args: ArticleAction) => ArticleAction;
