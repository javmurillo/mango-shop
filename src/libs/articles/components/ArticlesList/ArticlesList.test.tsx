import { cleanup, render } from '@testing-library/react';
import TestRenderer from 'react-test-renderer';
import { Article as ArticleDto } from '../../models/article';
import { ArticlesList } from './ArticlesList';

afterEach(cleanup);

const setup = () => {
  const utils = render(
    <ArticlesList articlesList={testArticles} ariaLabel="articles-list" />
  );
  const input = utils.getByLabelText('articles-list');
  return {
    input,
    ...utils,
  };
};

describe('<ArticlesList/> tests', () => {
  test('renders correctly', () => {
    const { input } = setup();
    expect(input).toBeInTheDocument();
  });

  test('shoud match snapshot', () => {
    const tree = TestRenderer.create(
      <ArticlesList articlesList={testArticles} />
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});

const testArticles: ArticleDto[] = [
  {
    id: '1',
    name: 'Test article',
    currentPrice: 75,
    oldPrice: 90,
    description: 'Test description',
    img: 'Test image',
  },
  {
    id: '2',
    name: 'Test article',
    currentPrice: 50,
    oldPrice: 70,
    description: 'Test description',
    img: 'Test image',
  },
  {
    id: '3',
    name: 'Test article',
    currentPrice: 25,
    oldPrice: 50,
    description: 'Test description',
    img: 'Test image',
  },
];
