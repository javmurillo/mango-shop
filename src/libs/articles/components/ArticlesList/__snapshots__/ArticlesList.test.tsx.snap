// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArticlesList/> tests shoud match snapshot 1`] = `
<div
  className="sc-jSgupP brmkFP row"
>
  <div
    aria-label="article-card"
    className="sc-hKgILt iRatjO card"
  >
    <img
      className="card-img-top"
      src="Test image"
    />
    <div
      className="card-body"
    >
      <div
        className="card-title h5"
      >
        Test article
      </div>
      <p
        className="card-text"
      >
        Test description
      </p>
    </div>
    <div
      className="sc-eCssSg bUELJj card-footer"
    >
      <span
        className="badge badge-success"
      >
        Deal of the day!
      </span>
    </div>
    <div
      className="text-muted card-footer"
    >
      <div
        className="sc-bdfBwQ dHJFwu"
      >
        <span
          className="sc-dlfnbm hsBRpN"
        >
          90
           €
        </span>
        <span
          className="sc-gsTCUz dACZDE"
        >
          75
           €
        </span>
      </div>
    </div>
  </div>
  <div
    aria-label="article-card"
    className="sc-hKgILt iRatjO card"
  >
    <img
      className="card-img-top"
      src="Test image"
    />
    <div
      className="card-body"
    >
      <div
        className="card-title h5"
      >
        Test article
      </div>
      <p
        className="card-text"
      >
        Test description
      </p>
    </div>
    <div
      className="sc-eCssSg bUELJj card-footer"
    >
      <span
        className="badge badge-success"
      >
        Deal of the day!
      </span>
    </div>
    <div
      className="text-muted card-footer"
    >
      <div
        className="sc-bdfBwQ dHJFwu"
      >
        <span
          className="sc-dlfnbm hsBRpN"
        >
          70
           €
        </span>
        <span
          className="sc-gsTCUz dACZDE"
        >
          50
           €
        </span>
      </div>
    </div>
  </div>
  <div
    aria-label="article-card"
    className="sc-hKgILt iRatjO card"
  >
    <img
      className="card-img-top"
      src="Test image"
    />
    <div
      className="card-body"
    >
      <div
        className="card-title h5"
      >
        Test article
      </div>
      <p
        className="card-text"
      >
        Test description
      </p>
    </div>
    <div
      className="sc-eCssSg bUELJj card-footer"
    >
      <span
        className="badge badge-success"
      >
        Deal of the day!
      </span>
    </div>
    <div
      className="text-muted card-footer"
    >
      <div
        className="sc-bdfBwQ dHJFwu"
      >
        <span
          className="sc-dlfnbm hsBRpN"
        >
          50
           €
        </span>
        <span
          className="sc-gsTCUz dACZDE"
        >
          25
           €
        </span>
      </div>
    </div>
  </div>
</div>
`;
